src/vs/code/electron-main/window.ts

```diff
@@ -426,6 +430,11 @@ export class VSCodeWindow {
 		delete configuration.filesToCreate;
 		delete configuration.filesToDiff;
 
+		// Update untitled files to restore so they come through in the reloaded window
```

Comment:
@Tyriar I am thinking more and more that untitled editors with backups should not be loaded from the main side but rather from the window itself. Files for example restore because we store their state into local storage (via `IEditorInputFactory` -  see https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/common/editor/editorStacksModel.ts#L621).

If we add a factory for untitled editors, we would just rely on that to bring the untitled back on reopen. It would also fix the issue that untitled would reopen at the right location where they where before.